osf/migrations/0077_auto_20180116_1322.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.7 on 2018-01-16 19:22
+from __future__ import unicode_literals
+
+from framework.auth import Auth
+from osf.models import AbstractNode, Comment, Guid
```

Comment:
Prefer `state.get_model` to direct imports in migrations

Reply:
I am having a hard time getting a model loaded this way to have all the same functionality as a direct import.

Reply:
What issues are you having?

I only recommend this because it will break down the road when imported models change -- the import causes Django to try to read/write them to SQL as they exist at runtime, rather than as the models/tables exist at this point in the migrations.

Reply:
oh ok, I will work more at this.  I couldn't seem to call NodeWikiPage.load(...) or NodeWikiPage.objects.get(guids___id=...).  Seems important I get this working though, so will try again!

Reply:
@mfraezz, if I do `NodeWikiPage = state.get_model('addons_wiki', 'nodewikipage')`, I can't load specific items.  Have you seen this before?

NodeWikiPage.objects.get(guids___id=version_list[0]) ERROR
 (django.core.exceptions.FieldError: Cannot resolve keyword 'guids' into field.) 

NodeWikiPage.load(version_list[-1]) ERROR
(type object 'NodeWikiPage' has no attribute 'load')

Reply:
Ah, right, this problem. The gist of it is that Django creates these models dynamically, and mixed-in aspects of the models aren't... mixed-in, and custom model managers aren't used. [This part](https://docs.djangoproject.com/en/2.0/topics/migrations/#historical-models) of the documentation touches on it a bit.

A valid workaround to look something up by its `Guid`, since `guids` and `GuidMixinQueryset` are both not availabe, should be to look the `Guid` itself up and then go to the referent object.

\<whisper> But SQL would be faster \</whisper>

Reply:
ah, I understand @mfraezz, thank you!

<whisper> You are right about SQL, but I've reduced the time to just a few minutes on a staging dump with bulk create/update.  I think it's worth testing on a prod dump before investing in a rewrite..</whisper>

Reply:
That's fair. Let me know when it's ready, and I can time it